src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -50,8 +51,16 @@ public class LibraryDependency : IEquatable<LibraryDependency>
         /// <summary>Initializes a new instance of the LibraryDependency class.</summary>
         /// <remarks>Required properties must be set when using this constructor.</remarks>
         public LibraryDependency()
+            : this(new List<NuGetLogCode>())
         {
-            NoWarn = new List<NuGetLogCode>();
+        }
+
+        /// <summary>Initializes a new instance of the LibraryDependency class with the specified NoWarn codes.</summary>
+        /// <param name="noWarn">Specifies a <see cref="List{T}" /> containing <see cref="NuGetLogCode" /> values.</param>
+        /// <remarks>Required properties must be set when using this constructor.</remarks>
+        public LibraryDependency(IList<NuGetLogCode> noWarn)
```

Comment:
I'm wondering if we can just require LibraryRange and NoWarn to be set.

Reply:
I'm not sure if its worth it.  By default the object will have an empty list that callers can populate.  Our codebase will always specify it.  Would it be a breaking change to make NoWarn `required`?